main/process-tx.zkasm

```diff
@@ -68,9 +70,34 @@ endCheckChainId:
                                         :CALL(isColdAddress) ; add tx.origin to touched addresses
         0                               :MSTORE(depth) ; Initial depth is 0
 
-;; Set gasPrice global var
+;; Set gasPrice global var depending on effectivePercentage [0-255] -> txGasPrice = Floor((gasPrice * (effectivePercentage + 1)) / 256)
+        ; _effGasPriceShifted = gasPrice * (effectivePercentage + 1)
+        ; A => gasPrice
         $ => A                          :MLOAD(txGasPriceRLP)
-        A                               :MSTORE(txGasPrice)
+        ; B => effectivePercentage
+        $ => B                          :MLOAD(effectivePercentageRLP)
+        ; B => [0, 255]
+        B + 1 => B
+        ; A*B + C = D * 2**256 + op(E)
+        0 => C
+        ; _effGasPriceShifted = gasPrice * (effectivePercentage + 1)
+        $${var _effGasPriceShifted = A * B}
+        ; get value above 256 bits
+        ${_effGasPriceShifted >> 256} => D
+        ; compute ARITH
+        ${_effGasPriceShifted} => E              :ARITH
+
+        ; txGasPrice = _effGasPriceShifted / 256
+        256 => B
+        ; (_effGasPriceShifted / 256)(A) * 256(B) + (_effGasPriceShifted % 256)(C) = D * 2**256 + op(E)
+        ${_effGasPriceShifted / 256} => A        :MSTORE(txGasPrice)
+        ${_effGasPriceShifted % 256} => C
+        ; compute ARITH
+        E                                       :ARITH
```

Comment:
Maybe I still don't quite understand how this works, so correct where if I'm wrong. 

I believe the purpose of this ARITH is to contstrain the free calculations above (`_effGasPriceShifted / 256` and `_effGasPriceShifted % 256`)

This `ARITH` line calculates `AB + C` and updates `E` and `D`.

What would stop malicious prover to return arbitrary value into `A` instead of `_effGasPriceShifted / 256`? This ARITH doesn't seem to constrain neither of registers values (unlike the first ARITH above?)
(for `C` it is more clear, it is at least checked for <256 afterwards)